pkg/operator/controller/route-metrics/controller.go

```diff
@@ -0,0 +1,248 @@
+package routemetrics
+
+import (
+	"context"
+	"fmt"
+
+	logf "github.com/openshift/cluster-ingress-operator/pkg/log"
+
+	corev1 "k8s.io/api/core/v1"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/cluster"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+
+	operatorv1 "github.com/openshift/api/operator/v1"
+	routev1 "github.com/openshift/api/route/v1"
+)
+
+const (
+	controllerName = "route_metrics_controller"
+)
+
+var (
+	log = logf.Logger.WithName(controllerName)
+)
+
+// New creates the route metrics controller. This is the controller
+// that handles all the logic for gathering and exporting
+// metrics related to route resources.
+func New(mgr manager.Manager) (controller.Controller, error) {
+	// Create a new cluster with a new cache to watch on Route objects from every namespace.
```

Comment:
```suggestion
	// Create a copy of the cluster, but with a new cache to watch on Route objects from every namespace.
```

Reply:
I consulted with @Miciah on this, and it seems like a good approach, though it is novel.  If you have a reference to an established or documented pattern, please cite it here.  The approach could use some more explanation, please.

Reply:
I couldn't find much documentation regarding ```NewKindWithCache```. Found this issue https://github.com/kubernetes-sigs/controller-runtime/issues/1824 which led me to K8ssandra Operator implementation. From there I understood how a separate cluster is needed to be created for ```NewKindWithCache```. Few places of reference in the K8ssandra Operator code:
https://github.com/k8ssandra/k8ssandra-operator/blob/main/controllers/k8ssandra/k8ssandracluster_controller.go#L194-L201
https://github.com/k8ssandra/k8ssandra-operator/blob/main/controllers/config/clientconfig_controller.go#L163-L187